example/src/useAxios/RaceCondition.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import Container from '../Container';
+import Heading from '../Heading';
+import TextBlock from '../TextBlock';
+import { useAxios } from '../../../src';
+
+interface Data {
+  data: {
+    name: string;
+    color: string;
+  };
+}
+
+export default () => {
+  const colors = {
+    CERULEAN: 1,
+    FUCHSIA: 2,
+  };
+  const [delay, setDelay] = useState(3000);
+  const [thing, setThing] = useState(colors.CERULEAN);
+
+  const { data, error, loading } = useAxios<Data>(
+    `http://slowwly.robertomurray.co.uk/delay/${delay}/url/https://reqres.in/api/things/${thing}`
+  );
+
+  setTimeout(() => {
+    setDelay(1000);
+    setThing(colors.FUCHSIA);
```

Comment:
`setTimeout` should always be wrapped by a `useEffect` - what are you trying to do here?

Reply:
Removed `setTimeout` and wrapped the state changes in a `useEffect`